crates/boundless-market/src/client.rs

```diff
@@ -470,6 +470,38 @@ where
             self.set_verifier.fetch_receipt_with_claim(seal, claim, journal.to_vec()).await?;
         Ok((journal, receipt))
     }
+
+    /// Fetch an order as a proof request and signature pair.
+    ///
+    /// If the request is not found in the boundless market, it will be fetched from the order stream service.
+    pub async fn fetch_order(
+        &self,
+        request_id: U256,
+        tx_hash: Option<B256>,
+    ) -> Result<Order, ClientError> {
+        match self.boundless_market.get_submitted_request(request_id, tx_hash).await {
+            Ok((request, signature_bytes)) => Ok(Order {
+                request,
+                signature: PrimitiveSignature::try_from(signature_bytes.as_ref())
+                    .map_err(|_| ClientError::Error(anyhow!("Failed to parse signature")))?,
+            }),
+            Err(_) => {
+                let orders = self
+                    .offchain_client
+                    .as_ref()
+                    .context(
+                        "Order stream client not available. Please provide an order stream URL",
+                    )?
+                    .fetch_order(request_id)
+                    .await?;
+                if orders.is_empty() {
+                    return Err(ClientError::Error(anyhow!("Order not found")));
+                }
+                // TODO: Handle multiple orders
+                Ok(orders[0].clone())
```

Comment:
Not sure what's best here: there might be the chance that multiple orders share the same request_id.
Ideally one would identify a request with its digest and use that to request the order. However we do not currently have enough tooling for make this easy.

Reply:
Hmm for now having an optional `requestDigest` in the fulfill command probably makes the most sense. Then if  there are multiple orders, and they haven't provided `requestDigest` we error out?

Reply:
Might also be worth discussing if we really need the multiple orders single request id property. Its definitely interesting, but its not simple or intuitive for users, and if its passing on this type of complexity to users I'm not 100% sure its worth it.

(it also makes the contracts quite a bit more complex)

Reply:
Ok IO've refactored a bit with [fc8fde7](https://github.com/boundless-xyz/boundless/pull/362/commits/fc8fde7ce3c260a7b9bcdf33d851198707947c44)